pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -270,15 +329,26 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByChildSubnet(subnets[0])
 		if len(bms) > 0 {
 			dependency := r.SubnetBindingService.GetSubnetConnectionBindingMapCRName(bms[0])
-			return nil, &errorWithRetry{
+			return nil, subnetCR, &errorWithRetry{
 				message: fmt.Sprintf("Target Subnet CR %s is associated by %s", name, dependency),
 				error:   fmt.Errorf("target Subnet %s is already associated by SubnetConnectionBindingMap %s", name, dependency),
 				retry:   true,
 			}
 		}
 	}
 
-	return subnets, nil
+	// Check if the Subnet CR is for pre-created Subnet
+	if isTarget {
+		if subnetCR.Status.Shared {
```

Comment:
Can we move this check to function `func (r *Reconciler) validateDependency` within the condition `if bindingMap.Spec.TargetSubnetName != ""`

Reply:
Updated, thanks!